src/pyff/resource.py

```diff
@@ -177,6 +178,50 @@ def to_dict(self) -> Dict[str, Any]:
         return res
 
 
+class ResourceLoadState(str, Enum):
+    Fetched = 'Fetched'
+    Parsing = 'Parsing'
+    Parsed = 'Parsed'
+    Ready = 'Ready'
+
+
+class ResourceInfo(BaseModel):
+    resource: str  # URL
+    state: Optional[ResourceLoadState] = None
+    http_headers: Dict[str, Any] = Field({})
+    reason: Optional[str] = None
+    status_code: Optional[str]  # HTTP status code as string. TODO: change to int
+    parser_info: Optional[ParserInfo] = None
+    expired: Optional[bool] = None
+    exception: Optional[BaseException] = None
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def to_dict(self):
```

Comment:
Its reasonable to ask if this "human readable json" stuff is really a good idea going forward